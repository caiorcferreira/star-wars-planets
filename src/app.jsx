import React from "react";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";

import "../node_modules/normalize.css/normalize.css";

import App from "./components/app";

ReactDOM.render(<App />, document.getElementById("app"));
