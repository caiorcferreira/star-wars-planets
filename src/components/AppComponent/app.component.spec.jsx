import React from "react";
import { shallow } from "enzyme";

import { App } from "./app.component";

import { planetFixture } from "../../tests/fixture/planet.fixture";

describe("Application component", () => {
  let fetchPlanet, planet;
  beforeEach(() => {
    fetchPlanet = jest.fn();
    planet = {
      data: planetFixture,
      loading: false,
      error: null
    };
  });

  it("should render component with planet", () => {
    const wrapper = shallow(<App fetchPlanet={fetchPlanet} planet={planet} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should render component loading", () => {
    const wrapper = shallow(
      <App fetchPlanet={fetchPlanet} planet={{ ...planet, loading: true }} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should render error", () => {
    const wrapper = shallow(
      <App
        fetchPlanet={fetchPlanet}
        planet={{ ...planet, error: "Not found" }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
