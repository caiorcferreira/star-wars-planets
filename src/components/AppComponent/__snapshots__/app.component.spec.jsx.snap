// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application component should render component loading 1`] = `
<main
  className="container"
>
  <h1
    className="title"
  >
    Star Wars Planets
  </h1>
  <div
    className="box center"
  >
    <Loader />
  </div>
</main>
`;

exports[`Application component should render component with planet 1`] = `
<main
  className="container"
>
  <h1
    className="title"
  >
    Star Wars Planets
  </h1>
  <div
    className="box"
  >
    <div>
      <div
        className="planet-info"
      >
        <h2
          className="planet-info__name"
        >
          Tatooine
        </h2>
        <hr />
        <p
          className="planet-info__population"
        >
          Population: 
          <span>
            200000
          </span>
        </p>
        <p
          className="planet-info__climate"
        >
          Climate:
           
          <span>
            Arid
          </span>
        </p>
        <p
          className="planet-info__terrain"
        >
          Terrain:
           
          <span>
            Desert
          </span>
        </p>
        <p
          className="planet-info__filmsFeatured"
        >
          Featured In 
          <span>
            5
          </span>
           films
        </p>
      </div>
      <div
        className="center"
      >
        <button
          className="planet-info__button"
          onClick={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
            }
          }
        >
          Next
        </button>
      </div>
    </div>
  </div>
</main>
`;

exports[`Application component should render error 1`] = `
<main
  className="container"
>
  <h1
    className="title"
  >
    Star Wars Planets
  </h1>
  <div
    className="box"
  >
    <p>
      Not found
    </p>
    <div>
      <div
        className="planet-info"
      >
        <h2
          className="planet-info__name"
        >
          Tatooine
        </h2>
        <hr />
        <p
          className="planet-info__population"
        >
          Population: 
          <span>
            200000
          </span>
        </p>
        <p
          className="planet-info__climate"
        >
          Climate:
           
          <span>
            Arid
          </span>
        </p>
        <p
          className="planet-info__terrain"
        >
          Terrain:
           
          <span>
            Desert
          </span>
        </p>
        <p
          className="planet-info__filmsFeatured"
        >
          Featured In 
          <span>
            5
          </span>
           films
        </p>
      </div>
      <div
        className="center"
      >
        <button
          className="planet-info__button"
          onClick={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
            }
          }
        >
          Next
        </button>
      </div>
    </div>
  </div>
</main>
`;
